docs/concepts/cluster-administration/cloud-providers.md

```diff
@@ -58,50 +58,108 @@ The information for the annotations for AWS is taken from the comments on [aws.g
 
 # OpenStack
 This section describes all the possible configurations which can
-be used when using OpenStack with Kubernetes.
+be used when using OpenStack with Kubernetes. The OpenStack cloud provider
+implementation for Kubernetes supports the use of these OpenStack services from
+the underlying cloud, where available:
+
+Service                  | API Version(s) | Required
+-------------------------+----------------+----------
+Block Storage (Cinder)   | V1†, V2        | No
+Compute (Nova)           | V2             | No
+Identity (Keystone)      | V2, V3         | Yes
+Load Balancing (Neutron) | V1‡, V2        | No
+
+† Block Storage V1 API support is deprecated, support for Block Storage V3 will
+  be added in the future.
+‡ Load Balancing V1 API support is deprecated.
+
+Service discovery is achieved by listing the service catalog managed by
+OpenStack Identity (Keystone) using the `auth-url` provided in the provider
+configuration. The provider will gracefully degrade in functionality when
+OpenStack services other than Keystone are not available and simply disclaim
+support for impacted features. Certain features are also enabled or disabled
+based on the list of extensions published by Neutron in the underlying cloud.
 
 ## cloud.conf
 Kubernetes knows how to interact with OpenStack via the file cloud.conf. It is the file that will provide Kubernetes with credentials and location for the OpenStack auth endpoint.
 You can create a cloud.conf file by specifying the following details in it
 
-### Minimal configuration
-This is an example of a minimal configuration that touches the values that most often need to be set:
+### Typical configuration
+This is an example of a typical configuration that touches the values that most
+often need to be set. It points the provider at the OpenStack cloud's Keystone
+endpoint, provides details for how to authenticate with it, and configures the
+load balancer:
 
 ```yaml
 [Global]
-username=user  
-password=pass  
-auth-url=https://<keystone_ip>/identity/v3  
-tenant-id=c869168a828847f39f7f06edd7305637  
+username=user
+password=pass
+auth-url=https://<keystone_ip>/identity/v3
+tenant-id=c869168a828847f39f7f06edd7305637
 domain-id=2a73b8f597c04551a0fdc8e95544be8a
 
 [LoadBalancer]
-subnet-id=6937f8fa-858d-4bc9-a3a5-18d2c957166a  
+subnet-id=6937f8fa-858d-4bc9-a3a5-18d2c957166a
 ```
 
 #### Global
```

Comment:
I suppose that's possible to use `username` or `user-id`,  `tenant-name` or `tenant-id`, `domain-name` or `domain-id`. The `trust-id`, `region` and `ca-file` parameters are missing, @dims do you know what these parameters are about?

Reply:
A **region** is a general division of an OpenStack deployment. 
Although a region does not have a geographical connotation, a deployment can use a geographical name for a region ID, such as us- east. `/v3/regions` (Keystone)
A **trust** represents a user’s (the trustor) authorization to delegate roles to another user (the trustee), and optionally allow the trustee to impersonate the trustor.  `/v3/OS-TRUST/trusts` (Keystone)

Reply:
Added `region` and `trust`, `ca-file` still outstanding.

Reply:
A region can be composed of multiple availability zones and this information is essential for Federation. 
Once the cluster nodes are labeled (automatically) `failure-domain.beta.kubernetes.io/zone=myzone` and it's needed the label `failure-domain.beta.kubernetes.io/region=myregion` too. 
It's possible to have multiple availability zones and only one region per cluster. 

The Cluster API object (federation) uses the information (Region and Zones) in the ClusterStatus.
https://github.com/kubernetes/federation/blob/master/federation/apis/federation/v1beta1/types.go#L86-L92


Suppose you have a federation with this characteristics:

Federation Name: `fed` created in `mydomain.com` 

|   Clusters   |  Cloud Provider  |  Region     | Zone          |
|:------------:|:----------------:|:-----------:|:-------------:|
| fr-cluster   | OpenStack        | eu-west-2   | eu-france     |
| uk-cluster   | AWS              | eu-west-2   | eu-london     |
| usa-cluster  | Google Cloud     | us-central1 | us-central1-a |
| chn-cluster  | Azure            | east-asia   | xangai        |

Creating a federated service `foo` in the namespace `bar`
The DNS entries in de DNS provider of federation has the format:

foo.bar.fed.svc.mydomain.com 
foo.bar.fed.svc.eu-west-2.mydomain.com 
foo.bar.fed.svc.eu-west-2.eu-france.mydomain.com 
foo.bar.fed.svc.eu-west-2.eu-london.mydomain.com 
foo.bar.fed.svc.us-central1.mydomain.com 
foo.bar.fed.svc.us-central1.us-central1-a.mydomain.com 
...

If the `foo` service is unavailable,  a new CNAME entry is created pointing to a next top-level domain, e.g:

* `foo` service is down in the `eu-france` **zone**
`foo.bar.fed.svc.eu-west-2.eu-france.mydomain.com` > `foo.bar.fed.svc.eu-west-2.mydomain.com`

* `foo` service is down in the `eu-west-2` **region** :
`foo.bar.fed.svc.eu-west-2.mydomain.com` > `foo.bar.fed.svc.mydomain.com`